src/compiler/sys.ts

```diff
@@ -294,6 +294,50 @@ namespace ts {
                 };
             }
 
+            function createWatchedFileSet() {
+                const watchedDirectories: { [path: string]: FileWatcher } = {};
+                const watchedFiles: { [fileName: string]: (fileName: string, removed?: boolean) => void; } = {};

```

Comment:
Actually, consider creating a local type alias named`FileChangedCallback` type and use it here and for the `callback` parameter of `addFile`